src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -1224,4 +1229,25 @@ public static void onCreativeModeTabBuildContents(CreativeModeTab tab, CreativeM
         for (var entry : entries)
             output.accept(entry.getKey(), entry.getValue());
     }
+    
+    // Runs during Minecraft construction, before initial resource loading.
+    @ApiStatus.Internal
```

Comment:
Not sure it is worth it but we should consider if we want to *enforce* this only being called by forge as I believe it is only called once so just setting a boolean after first call and then throwing for any additional calls would prevent any mods from being able to improperly call this.